diff -Naur perl-5.16.0.orig/installperl perl-5.16.0/installperl
--- perl-5.16.0.orig/installperl	2012-04-25 08:18:34.000000000 +0800
+++ perl-5.16.0/installperl	2012-07-29 18:52:24.196551860 +0800
@@ -569,43 +569,43 @@
 # Also skip $mainperl if the user opted to have it be a link to the
 # installed perl.
 
-if (!$versiononly && $otherperls) {
-    my ($path, @path);
-    my $dirsep = ($Is_OS2 || $Is_W32 || $Is_NetWare) ? ';' : ':' ;
-    ($path = $ENV{"PATH"}) =~ s:\\:/:g ;
-    @path = split(/$dirsep/, $path);
-    if ($Is_VMS) {
-	my $i = 0;
-	while (exists $ENV{'DCL$PATH' . $i}) {
-	    my $dir = unixpath($ENV{'DCL$PATH' . $i});  $dir =~ s-/$--;
-	    push(@path,$dir);
-	}
-    }
-    my @otherperls;
-    my %otherperls;
-    for (@path) {
-	next unless m,^/,;
-	# Use &samepath here because some systems have other dirs linked
-	# to $mainperldir (like SunOS)
-	next unless -d;
-	next if samepath($_, $binexp);
-	next if samepath($_, cwd());
-	next if ($mainperl_is_instperl && samepath($_, $mainperldir));
-	my $otherperl = "$_/$perl$exe_ext";
-	next if $otherperls{$otherperl}++;
-	push(@otherperls, $otherperl)
-	    if (-x $otherperl && ! -d $otherperl);
-    }
-    if (@otherperls) {
-	warn "\nWarning: $perl appears in your path in the following " .
-	    "locations beyond where\nwe just installed it:\n";
-	for (@otherperls) {
-	    warn "    ", $_, "\n";
-	}
-	warn "\n";
-    }
-
-}
+#if (!$versiononly && $otherperls) {
+#   my ($path, @path);
+#   my $dirsep = ($Is_OS2 || $Is_W32 || $Is_NetWare) ? ';' : ':' ;
+#   ($path = $ENV{"PATH"}) =~ s:\\:/:g ;
+#   @path = split(/$dirsep/, $path);
+#   if ($Is_VMS) {
+#	my $i = 0;
+#	while (exists $ENV{'DCL$PATH' . $i}) {
+#	    my $dir = unixpath($ENV{'DCL$PATH' . $i});  $dir =~ s-/$--;
+#	    push(@path,$dir);
+#	}
+#   }
+#   my @otherperls;
+#   my %otherperls;
+#   for (@path) {
+#	next unless m,^/,;
+#	# Use &samepath here because some systems have other dirs linked
+#	# to $mainperldir (like SunOS)
+#	next unless -d;
+#	next if samepath($_, $binexp);
+#	next if samepath($_, cwd());
+#	next if ($mainperl_is_instperl && samepath($_, $mainperldir));
+#	my $otherperl = "$_/$perl$exe_ext";
+#	next if $otherperls{$otherperl}++;
+#	push(@otherperls, $otherperl)
+#	    if (-x $otherperl && ! -d $otherperl);
+#   }
+#   if (@otherperls) {
+#	warn "\nWarning: $perl appears in your path in the following " .
+#	    "locations beyond where\nwe just installed it:\n";
+#	for (@otherperls) {
+#	    warn "    ", $_, "\n";
+#	}
+#	warn "\n";
+#   }
+#
+#}
 
 $packlist->write() unless $opts{notify};
 print "  Installation complete\n" if $opts{verbose};
