--- e2fsprogs-1.42.2.orig/doc/Makefile.in	2010-02-11 07:16:03.000000000 +0800
+++ e2fsprogs-1.42.2/doc/Makefile.in	2012-05-16 11:18:14.317030402 +0800
@@ -18,7 +18,7 @@
 
 all:: libext2fs.info libext2fs.dvi
 
-install-doc-libs: libext2fs.info libext2fs.dvi
+install-doc-libs: libext2fs.info
 	$(Q) $(RM) -rf $(DESTDIR)$(infodir)/libext2fs.info*
 	$(E) "	MKINSTALLDIRS $(infodir)"
 	$(Q) $(MKINSTALLDIRS) $(DESTDIR)$(infodir)
